File: NeuroflowLib/activation_description.cpp
#include "stdafx.h"
#include "activation_description.h"

USING

activation_description::activation_description(activation_function function, float alpha) :
_function(function),
_alpha(alpha)
{
}

activation_function activation_description::function() const
{
    return _function;
}

float activation_description::alpha() const
{
    return _alpha;
}

File: _vault/prev_version_win/Neuroflow/NeuralNetworks/ActivationDescription.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Neuroflow.NeuralNetworks
{
    public sealed class ActivationDescription : LayerDescription
    {
        public ActivationDescription() :
            this(ActivationFunction.Sigmoid)
        {
        }
        public ActivationDescription(ActivationFunction function) :
            this(function, function == ActivationFunction.Sigmoid ? 1.75f : 1.0f)
        {
        }

        public ActivationDescription(ActivationFunction function, float alpha)
        {
            Function = function;
            Alpha = alpha;
        }

        public ActivationFunction Function { get; set; }

        public float Alpha { get; set; }
    }
}
